src/main/kotlin/com/figure/gradle/semver/internal/tasks/CreateAndPushVersionTagTask.kt

```diff
@@ -39,6 +40,11 @@ abstract class CreateAndPushVersionTagTask : DefaultTask() {
                 .build()
         )
 
+        val tags = git.tagList().call()
+        val tagAlreadyExists = versionTagName.get() in tags.map { it.name.replace("refs/tags/", "") }
+
+        if (tagAlreadyExists) throw TagAlreadyExistsException(versionTagName.get())
+
```

Comment:
This should make it clearer in Github actions for everyone rather than letting the Git exception bubble up and show something confusing.